src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -0,0 +1,512 @@
+
+        private void onAmqpHeader(AmqpHeaderFW header)
+        {
+            doAmqpHeader(header.buffer(), header.offset(), header.limit());
+        }
+
+        private void onAmqpOpen(
+            AmqpFrameFW open)
+        {
+            // TODO
+        }
+
+        private void onAmqpBegin(
+            AmqpFrameFW begin)
+        {
+            // TODO
+            /*
+            * get a session from map with channel
+            * */
+        }
+
+        private void onAmqpAttach(
+            AmqpFrameFW attach)
+        {
+            // TODO
+        }
+
+        private void onAmqpFlow(
+            AmqpFrameFW flow)
+        {
+            // TODO
+        }
+
+        private void onAmqpTransfer(
+            AmqpFrameFW transfer)
+        {
+            // TODO
+        }
+
+        private void onAmqpDisposition(
+            AmqpFrameFW disposition)
+        {
+            // TODO
+        }
+
+        private void onAmqpDetach(
+            AmqpFrameFW detach)
+        {
+            // TODO
+        }
+
+        private void onAmqpEnd(
+            AmqpFrameFW end)
+        {
+            // TODO
+        }
+
+        private void onAmqpClose(
+            AmqpFrameFW close)
+        {
+            // TODO
+        }
+
+        private int decodeHeader(
+            final DirectBuffer buffer,
+            final int offset,
+            final int length)
+        {
+            final AmqpHeaderFW amqpHeader = amqpHeaderRO.wrap(buffer, offset, offset + length);
+            onAmqpHeader(amqpHeader);
+            decodeState = this::decodeFrame;
+            return amqpHeader.sizeof(); // should be 8
+        }
+
+        // TODO
+        private int decodeFrame(
+            final DirectBuffer buffer,
+            final int offset,
+            final int length)
+        {
+            int consumed;
+
+            final AmqpFrameFW amqpFrame = amqpFrameRO.wrap(buffer, offset, offset + length); // TODO
+
+            switch (amqpFrame.performative())
+            {
+                case 0x00:
+                    onAmqpOpen(amqpFrame);
+                    break;
+                case 0x01:
+                    onAmqpBegin(amqpFrame);
+                    break;
+                case 0x02:
+                    onAmqpAttach(amqpFrame);
+                    break;
+                case 0x03:
+                    onAmqpFlow(amqpFrame);
+                    break;
+                case 0x04:
+                    onAmqpTransfer(amqpFrame);
+                    break;
+                case 0x05:
+                    onAmqpDisposition(amqpFrame);
+                    break;
+                case 0x06:
+                    onAmqpDetach(amqpFrame);
+                    break;
+                case 0x07:
+                    onAmqpEnd(amqpFrame);
+                    break;
+                case 0x08:
+                    onAmqpClose(amqpFrame);
+                    break;
+            }
+            return amqpFrame.sizeof();
```

Comment:
Does `amqpFrame` wrap the entire frame, or just the generic frame header?
That will determine whether or not this return value is correct since you want the decode loop to skip this many bytes to advance to the beginning of the next AMQP frame when progressing through the network stream.